### PR TITLE
Add github authentication

### PR DESCRIPTION
**Pull request by FraDim**

### CODE REVIEW

File: config/puma.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 # Specifies the `port` that Puma will listen on to receive requests, default is 3000.
```
Comment (MiroslavCsonka) [inline config/puma.rb]: **minor detail** this should also probably be 3000

**Comment by FraDim** [comment]
Review please @LogaJ @MiroslavCsonka

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # frozen_string_literal: true
 class ApplicationController < ActionController::Base
+  # before_action :authenticate_user!
```
Comment (LogaJ) [inline app/controllers/application_controller.rb]: Oh Rly?

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module Users
+  class OmniauthCallbacksController < Devise::OmniauthCallbacksController
```
Comment (LogaJ) [inline app/controllers/users/omniauth_callbacks_controller.rb]: No need to mention omni auth, it's an implementation detail

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    def github
+      user = SaveAuthenticatedUser.new(github_arguments).call
```
Comment (LogaJ) [inline app/controllers/users/omniauth_callbacks_controller.rb]: Please `call!` it's much clearer that nothing below will be executed if the above fails.

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (LogaJ) [inline spec/models/user_spec.rb]: Oh Rly?

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+Devise.setup do |config|
+  require 'devise/orm/active_record'
+
+  config.skip_session_storage = [:http_auth]
+
+  config.sign_out_via = :delete
+
+  config.omniauth :github, ENV.fetch('GITHUB_CLIENT_ID'), ENV.fetch('GITHUB_CLIENT_SECRET'), scope: 'user,public_repo', callback_url: 'http://localhost:3000/users/auth/github/callback'
```
Comment (MiroslavCsonka) [inline config/initializers/devise.rb]: Do we need to override the `callback_url` in production?

**Pull request merged**